source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,277 @@
+                                             * std::cos(th)))));
+
+      /* Convert to degrees. */
+      const double lon_degrees = lon * (180 / numbers::PI);
+      const double lat_degrees = lat * (180 / numbers::PI);
+
+      /* Set all longitudes between [0,360]. */
+      if (lon_degrees < 0)
+        return std::make_pair(lat_degrees, lon_degrees + 360);
+      else if (lon_degrees > 360)
+        return std::make_pair(lat_degrees, lon_degrees - 360);
+      else
+        return std::make_pair(lat_degrees, lon_degrees);
+
+    }
+
+    template <>
+    double
+    AdiabaticBoundary<2>::get_isotherm_depth(const double,
+                                             const double) const
+    {
+      AssertThrow(false, ExcInternalError());
+      return 0;
+    }
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::get_isotherm_depth(const double latitude,
+                                               const double longitude) const
+    {
+      /**
+       * Loop over the entire array and see if we find a point
+       * that's within delta of what we're looking for. If this
+       * point is not within the delta then test if it is within twice
+       * as much as delta.
+       */
+      for (unsigned int i = 0; i <= depths_iso.size();)
+        if (std::fabs(latitude - latitudes_iso[i]) <= delta && std::fabs(longitude - longitudes_iso[i]) < delta)
+          {
+            return -depths_iso[i]*1000;
+          }
+        else if (std::fabs(latitude - latitudes_iso[i]) <= delta*2 && std::fabs(longitude - longitudes_iso[i]) < delta*2)
+          {
+            return -depths_iso[i]*1000;
+          }
+        else
+          i++;
+      Assert(false, ExcInternalError());
+      return 0;
+    }
+
+    /**
+     * Calculate distance from the center of the Earth to the surface of WGS84.
+     */
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::radius_WGS84(const double &theta) const
+    {
+      const double eccentricity    = 8.1819190842622e-2;
+      const double semi_major_axis = 6378137.0;
+      return semi_major_axis/std::sqrt(1- eccentricity * eccentricity
+                                       * std::sin(numbers::PI*theta/180)*std::sin(numbers::PI*theta/180));
+    }
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::initial_temperature (const Point<dim> &) const
+    {
+      Assert (false, ExcNotImplemented());
+      return 0;
+    }
+
+    /**
+     * Get the depth of the adiabatic boundary isotherm for the current lat/long.
+     * If above the isotherm use a linear interpolation to the surface.
+     * If below the isotherm use an increase of .5 degrees per kilometer.
+     */
+
+    template <>
+    double
+    AdiabaticBoundary<3>::initial_temperature (const Point<3> &position) const
+    {
+      const std::pair<double, double> lat_long = lat_long_from_xyz_WGS84(position);
+      const double depth                       = position.norm() - radius_WGS84(lat_long.first);
```

Comment:
Is this definition the reason that you have the minus in the get_isotherm_depth function? Can you simply change the definition to have a positive depth for both? I do not feel strongly about whether both depths are positive or negative, I would just opt for having the same sign on both.